dis_snek/models/discord/base.py

```diff
@@ -12,30 +14,41 @@
 __all__ = ["ClientObject", "DiscordObject"]
 
 
+# noinspection PyMethodOverriding
 @define(slots=False)
 class ClientObject(DictSerializationMixin):
     _client: "Snake" = field(metadata=no_export_meta)
 
     @classmethod
-    def _process_dict(cls, data: Dict[str, Any], client: "Snake") -> Dict[str, Any]:
+    def _process_dict(cls, data: Dict[str, Any], client: "Snake", **kwargs) -> Dict[str, Any]:
         return super()._process_dict(data)
 
     @classmethod
-    def from_dict(cls: Type[T], data: Dict[str, Any], client: "Snake") -> T:
-        data = cls._process_dict(data, client)
-        return cls(client=client, **cls._filter_kwargs(data, cls._get_init_keys()))
+    def from_dict(cls: Type[T], data: Dict[str, Any], client, **kwargs) -> T:
+        """
+        Process and converts dictionary data received from discord api to object class instance.
+
+        parameters:
+            data: The json data received from discord api.
+
+        """
+        return super().from_dict(data, client=client, **kwargs)
 
     @classmethod
-    def from_list(cls: Type[T], datas: List[Dict[str, Any]], client: "Snake") -> List[T]:
-        return [cls.from_dict(data, client) for data in datas]
+    def from_list(cls: Type[T], data_list: List[Dict[str, Any]], client, **kwargs) -> List[T]:
+        """
+        Process and converts list data received from discord api to object class instances.
+
+        parameters:
+            data: The json data received from discord api.
+
+        """
+        return super().from_list(data_list, client=client, **kwargs)
 
-    def update_from_dict(self, data) -> None:
-        data = self._process_dict(data, self._client)
-        for key, value in self._filter_kwargs(data, self._get_keys()).items():
-            # todo improve
-            setattr(self, key, value)
+    def update_from_dict(self: T, data: Dict[str, Any], **kwargs) -> T:
+        return super().update_from_dict(data, client=self._client, **kwargs)
 
 
 @define(slots=False)
 class DiscordObject(SnowflakeObject, ClientObject):
-    pass
```

Comment:
Why?

Reply:
must have forgotten to revert correctly after testing some methods, anyway I will check though once the changes are firmed up (its pr is still a draft)